spec/system/consumer/shopping/products_spec.rb

```diff
@@ -82,5 +93,59 @@
         end
       end
     end
+
+    describe "filtering" do
+      before do
+        exchange1.update_attribute :pickup_time, "monday"
+        add_variant_to_order_cycle(exchange1, variant)
+        add_variant_to_order_cycle(exchange1, variant2)
+      end
+
+      context "product taxonomies" do
+        before do
+          login_as user
+          visit edit_admin_enterprise_path(distributor)
+
+          within(".side_menu") do
+            click_link "Shop Preferences"
+          end
+          find(".select2-input").click
+          find(".select2-result-label", text: "Tricky Taxon").click
+          click_button 'Update'
+          expect(flash_message).to eq('Enterprise "Testing Farm" has been successfully updated!')
+
+          logout
```

Comment:
Setting is doesn't work?
```rb
distributor.preferred_shopfront_product_sorting_method = "by_category"
```

Reply:
Yup, I've tried it and the lines:

```
distributor.preferred_shopfront_product_sorting_method = "by_category"
distributor.preferred_shopfront_taxon_order = "#{taxon.name}"
```

do work :+1:

Reply:
Update: actually, to mock this correctly, we seem to need the **id** and not the **name** (as above):

`distributor.preferred_shopfront_taxon_order = "#{taxon.id}"`

If we don't, then we [get this error](https://github.com/openfoodfoundation/openfoodnetwork/pull/9083#issuecomment-1097010785), so I guess #9046 is occurring because setting the category in the frontend is not communicating this preference correctly :+1: